src/main/java/com/hubspot/jinjava/lib/tag/RawTag.java

```diff
@@ -33,6 +34,14 @@ public String interpret(TagNode tagNode, JinjavaInterpreter interpreter) {
     LengthLimitingStringBuilder result = new LengthLimitingStringBuilder(
       interpreter.getConfig().getMaxOutputSize()
     );
+    if (interpreter.getConfig().isPreserveRawTags()) {
+      result.append(renderNodeRaw(tagNode));
+      throw new PreservedRawTagException(
```

Comment:
Why throw exception here instead of 
```
return result.toString();
```
?

Reply:
After working with the code base a bit more, I agree that it is simpler to return rather than throw an exception, and call `interpreter.getContext().handlePreservedRawTag();` from here instead of in the `TagNode`